dpctl/tensor/libtensor/include/kernels/sorting/topk.hpp

```diff
@@ -525,12 +552,10 @@ sycl::event topk_radix_impl(sycl::queue &exec_q,
         const sycl::context &ctx = exec_q.get_context();
 
         using dpctl::tensor::alloc_utils::sycl_free_noexcept;
-        cgh.host_task([ctx, workspace, tmp_tp] {
-            sycl_free_noexcept(workspace, ctx);
-            sycl_free_noexcept(tmp_tp, ctx);
-        });
+        cgh.host_task([ctx, workspace] { sycl_free_noexcept(workspace, ctx); });
     });
 
+    workspace_owner.release();
```

Comment:
I think it would be best to release the pointer before calling free.

Release also returns the original pointer.

This can be refactored to:
```c++
    auto work_space_ptr = workspace_owner.release();
    sycl::event cleanup_ev = exec_q.submit([&](sycl::handler &cgh) {
        cgh.depends_on(write_topk_ev);
        const sycl::context &ctx = exec_q.get_context();

        using dpctl::tensor::alloc_utils::sycl_free_noexcept;
        cgh.host_task([ctx, work_space_ptr] { sycl_free_noexcept(work_space_ptr, ctx); });
    });
```

If we plan to generalize this, it might also be interesting to construct some kind of variadic template for releasing some number of `unique_ptr`s wrapping USM allocations. Then we could cut this redundant and frequently used `host_task` code out everywhere.